### DISCUSSION TITLE
Preview Component with Dry/Required args

### DISCUSSION THREAD

**Discussion by danielnc** [palkan/view_component-contrib#23] [Q&A] [answered]
I couldn't find a good example, so help is greatly appreciated

I have a simple component that has some required params:

```ruby
# frozen_string_literal: true

class Shared::Navbar::Avatar::Component < ApplicationViewComponent
  option :current_user
  option :resource_name
  option :avatar_size, default: proc { 40 }

  def user_signed_in?
    current_user&.user_signed_in? || false
  end

  def sign_in_path
    new_session_path(resource_name)
  end
end
```

This is the preview code I am using based on the default generator:
```ruby
# frozen_string_literal: true

class Shared::Navbar::Avatar::Preview < ApplicationViewComponentPreview
  def default
    render_component(Shared::Navbar::Avatar::Component.new(current_user: Member.new, resource_name: :member))
  end
end

```

Whenever I try to preview the component I am getting the following error, not sure how to use the default preview template(even if it's possible tbh) once I have some required params

```
Failed to infer a component from the preview: Shared::Navbar::Component: option 'current_user' is required
Source:
<div class="<%= container_class %>">
  <%- if component -%>
    <%= render component %>
  <%- else -%>
  Failed to infer a component from the preview: <%= error %>
  <%- end -%>
</div>
```

Not sure what I am doing wrong, or what I need to change

The component works outside the preview, on the app code...

**Comment by palkan** [comment, marked as answer]
It seems that there is some problem with `render_component` method.

I believe, the exception is raised when we try to build a default component instance here: https://github.com/palkan/view_component-contrib/blob/d96e3b74a4f764036a6ff77d351afc7fa2bb3116/lib/view_component_contrib/preview/base.rb#L50

The question is where do we lost the `component` passed to the [`render_with_template`](https://github.com/palkan/view_component-contrib/blob/d96e3b74a4f764036a6ff77d351afc7fa2bb3116/lib/view_component_contrib/preview/base.rb#L59) method along the way? Only debugging could help here 🤷‍♂️.

**Reply by danielnc** [reply]
Not sure what caused this in the first place, when I started to debug this now, after those couple of days, it is now working

Not sure if it was some cache I needed to clean on my side but it works, so the original question is moot 👎
